internal/gengapic/genrest.go

```diff
@@ -67,25 +76,78 @@ func (g *generator) genRESTMethods(serv *descriptor.ServiceDescriptorProto, serv
+		fqn := g.descInfo.ParentFile[serv].GetPackage() + "." + serv.GetName()
+		return fmt.Errorf("service %q is missing option google.api.default_host", fqn)
+	}
+
+	p("func default%sRESTClientOptions() []option.ClientOption {", servName)
+	p("  return []option.ClientOption{")
+	p("    internaloption.WithDefaultEndpoint(%q),", host)
+	p("    internaloption.WithDefaultMTLSEndpoint(%q),", generateDefaultMTLSEndpoint(host))
+	p("    internaloption.WithDefaultAudience(%q),", generateDefaultAudience(host))
+	p("    internaloption.WithDefaultScopes(DefaultAuthScopes()...),")
+	p("  }")
+	p("}")
+
+	return nil
+}
+
 func (g *generator) restClientUtilities(serv *descriptor.ServiceDescriptorProto, servName string, imp pbinfo.ImportSpec, hasRPCForLRO bool) {
-	// p := g.printf
-
-	// clientName := camelToSnake(serv.GetName())
-	// clientName = strings.Replace(clientName, "_", " ", -1)
-
-	// NOTE(dovs): enable this function only when we can verify that it works.
-	// If we leave it commented out, we can change it, and we don't have to
-	// guarantee correctness or stability.
-	//
-	// p("// New%sRESTClient creates a new %s rest client.", servName, clientName)
-	// g.serviceDoc(serv)
-	// p("func New%[1]sRESTClient(ctx context.Context, opts ...option.ClientOption) (*%[1]sClient, error) {", servName)
-	// p("    c := &%s{", servName)
-	// p("    }")
-	// p("")
-	// p("    return &%sClient{internal%[1]sClient: c, CallOptions: default%[1]sCallOptions()}, nil", servName)
-	// p("}")
+	p := g.printf
+	lowcaseServName := lowcaseRestClientName(servName)
+
+	clientName := camelToSnake(serv.GetName())
+	clientName = strings.Replace(clientName, "_", " ", -1)
+
+	p("// New%sRESTClient creates a new %s rest client.", servName, clientName)
+	g.serviceDoc(serv)
+	p("func New%[1]sRESTClient(ctx context.Context, opts ...option.ClientOption) (*%[1]sClient, error) {", servName)
+	p("    clientOpts := default%sRESTClientOptions()", servName)
+	p("    httpClient, endpoint, err := httptransport.NewClient(ctx, append(clientOpts, opts...)...)")
+	p("    if err != nil {")
+	p("        return nil, err")
+	p("    }")
+	p("")
+	p("    c := &%s{", lowcaseServName)
+	p("        endpoint: endpoint,")
+	p("        httpClient: httpClient,")
+	p("    }")
+	p("    c.setGoogleClientInfo()")
+	p("")
+	// TODO(dovs): make rest default call options
+	// TODO(dovs): set the LRO client
+	p("    return &%[1]sClient{internalClient: c, CallOptions: default%[1]sCallOptions()}, nil", servName)
+	p("}")
+	p("")
+
+	g.restClientOptions(serv, servName)
+
+	// setGoogleClientInfo method
+	p("// setGoogleClientInfo sets the name and version of the application in")
+	p("// the `x-goog-api-client` header passed on each request. Intended for")
+	p("// use by Google-written clients.")
+	p("func (c *%s) setGoogleClientInfo(keyval ...string) {", lowcaseServName)
+	p(`  kv := append([]string{"gl-go", versionGo()}, keyval...)`)
+	p(`  kv = append(kv, "gapic", versionClient, "gax", gax.Version, "rest", "UNKNOWN")`)
```

Comment:
Maybe use `0.0` rather than `UNKNOWN` to make it clear this is an explicit choice (that way our data pipelines can fill in UNKNOWN for missing values, say)

Reply:
Lets keep this as UNKNOWN for now. We have other automation that post processes this value into something more meaningful

Reply:
This post-processing happens before publication? OK. Like I say, once the library is in use I don't want to pollute the space of values we provide.

Reply:
Yup, it happens before the PR is raised to land the changes: https://github.com/googleapis/google-cloud-go/blob/master/internal/gapicgen/generator/gapics.go#L85-L97